pkg/engine/plugins.go

```diff
@@ -184,25 +183,31 @@ func gatherPluginsFromSnapshot(plugctx *plugin.Context, target *deploy.Target) (
 	return set, nil
 }
 
-// ensurePluginsAreInstalled inspects all plugins in the plugin set and, if any plugins are not currently installed,
+// EnsurePluginsAreInstalled inspects all plugins in the plugin set and, if any plugins are not currently installed,
 // uses the given backend client to install them. Installations are processed in parallel, though
-// ensurePluginsAreInstalled does not return until all installations are completed.
-func ensurePluginsAreInstalled(ctx context.Context, d diag.Sink,
-	plugins pluginSet, projectPlugins []workspace.ProjectPlugin,
```

Comment:
We need to keep support for `projectPlugins`.

Reply:
I see, I removed it because it was unused and wasn't sure how to get at it from the other callsite, I'll find a way.

Reply:
ah, all fixed up now